File: 2000-reverse-prefix-of-word/2000-reverse-prefix-of-word.c
char* reversePrefix(char* word, char ch) {
    int idx = -1;
    for (int i=0;i<strlen(word);i++) {
        if (word[i] == ch) {
            idx = i;
            break;
        }
    }
    if (idx > 0) {
        for (int i=0,j=idx; i <= j; i++, j--) { 
            char c = word[i]; 
            word[i] = word[j]; 
            word[j] = c; 
    }
        return word;
    }
    else {
        return word;
    }
}

File: 2000-reverse-prefix-of-word/2000-reverse-prefix-of-word.py
class Solution:
    def reversePrefix(self, word: str, ch: str) -> str:
        idx = -1
        for i in range(len(word)):
            if word[i] == ch:
                idx = i
                break
        if idx > 0:
            return "".join(reversed(word[:i+1])) + word[i+1:]
        else:
            return word